contrib/opentimelineio_contrib/adapters/advanced_authoring_format.py

```diff
@@ -537,18 +538,20 @@ def _find_timecode_track_start(track):
     aaf_metadata = track.metadata.get("AAF", {})
 
     # Is this a Timecode track?
-    if aaf_metadata.get("MediaKind") == "Timecode":
-        edit_rate = aaf_metadata.get("EditRate", "0")
-        fps = aaf_metadata.get("Segment", {}).get("FPS", 0)
+    if aaf_metadata.get("MediaKind") in {"Timecode", "LegacyTimecode"}:
+        edit_rate = Fraction(aaf_metadata.get("EditRate", "0"))
+        if edit_rate.denominator == 1:
+            rate = edit_rate.numerator
+        else:
+            rate = float(edit_rate)
         start = aaf_metadata.get("Segment", {}).get("Start", "0")
+        physical_track_number = aaf_metadata.get("PhysicalTrackNumber", 1)
```

Comment:
Is the default value here needed? Would PhysicalTrackNumber ever be missing?

Reply:
I don't expect it would be. We should probably be able to assume the same for `EditRate`, `Segment`, and `Start` too, right? I also noticed that defaulting `Start` to a string `"0"` didn't match the files I looked like that had integer values.

I can do something like:
```try:
    edit_rate = Fraction(aaf_metadata.get["EditRate"])
    start = aaf_metadata["Segment"]["Start"]
    physical_track_number = aaf_metadata["PhysicalTrackNumber"]
except KeyError:
    continue```

That way we'll do a best-effort without totally dying if we're missing something.

Reply:
Since we still have a ways to go in understanding AAF deeply, I think we should throw an exception if we find something we don't expect. That isn't the best thing for the user at that moment, but it forces us to stop, figure out what we misunderstood, and then fix it. So maybe
```
...
except KeyError:
  raise AAFAdapterError("Timecode missing PhysicalTrackNumber")
```